iidm/iidm-api/src/main/java/com/powsybl/iidm/network/TerminalRef.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.network;
+
+import java.util.Optional;
+
+import com.powsybl.commons.PowsyblException;
+
+/**
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
+ */
+public final class TerminalRef {
+
+    public enum Side {
+        ONE,
+        TWO,
+        THREE
+    }
+
+    private final String id;
+    private final Side side;
+
+    private TerminalRef(String id, Side side) {
+        this.id = id;
+        this.side = side;
+    }
+
+    public static TerminalRef create(String id) {
+        return new TerminalRef(id, null);
+    }
+
+    public static TerminalRef create(String id, Side side) {
+        return new TerminalRef(id, side);
+    }
+
+    public static Optional<TerminalRef.Side> getConnectableSide(Terminal terminal) {
+        Connectable<?> c = terminal.getConnectable();
+        if (c instanceof Injection) {
+            return Optional.empty();
+        } else if (c instanceof Branch) {
+            return Optional.of(toSide(((Branch) c).getSide(terminal)));
+        } else if (c instanceof ThreeWindingsTransformer) {
+            return Optional.of(toSide(((ThreeWindingsTransformer) c).getSide(terminal)));
+        } else {
+            throw new IllegalStateException("Unexpected Connectable instance: " + c.getClass());
+        }
+    }
+
+    // FIXME(Luma) This has been moved from TerminalRefXml,
+    // Makes sense to have it here,
+    // In the future ...
+    // a Terminal ref specified for a switch could be resolved to different terminals depending on the topology
+    // This could also be written as TerminalRef.create(id , side).resolve(network)
+    public static Terminal resolve(String id, String sideText, Network network) {
```

Comment:
This should be moved to `TerminalRefXml` I think, not here in the api.

Reply:
Resolving terminal references to actual Network terminals is not only needed for XML serialisation.

Having it here will allow to resolve an "original, precise" definition of the terminal (at the end of a switch) to the right Network terminal with the current topology, that may change with actions/modifications performed by the user.